### DISCUSSION TITLE
Not prompted for Credentials

### DISCUSSION THREAD

**Discussion by jimmarks** [briis/unifiprotect#280] [Q&A] [answered]
When I follow the configuration, I can load this integration, but I'm never prompted for credentials.

I am not running my Network NVR on the same VLan though as my Home Assistant. I do have MDNS enabled, so I should be able to see it, but I think maybe I'm not getting prompted for all of the credentials because it can't find the protect device. Is there a way to hard set the credentials and IP address of the Protect device?

**Comment by briis** [comment]
Just to clarify where you are in the process:

1. You have installed the Integration through HACS?
2. You then went to the Configuration->Integrations Page, and added the Integration?
3. Then nothing happens? You are not presented with a dialog asking for your UniFi Protect IP, local username and password?

**Comment by jimmarks** [comment]
Correct.  I can see the integration when I open the HACS dashboard and look at the integrations as well, but when I click on it, I get the details box, but nothing allows me to enter any IP, or U & P

There are also no errors that I see in the logs pertaining to the Unifi Protect integration

**Comment by briis** [comment]
So when you are on http://YOUR_HASS_IP:8123/config/integrations and click the `+ ADD INTEGRATION` and search for Unifi Protect, nothing happens?

**Comment by jimmarks** [comment]
No.  I load the integration:

![Unifiprotect](https://user-images.githubusercontent.com/17133893/129910574-1b75b53f-c205-4aeb-9a0e-4edd3c571a11.PNG)

But the popup asking for credentials never shows up, and there isn't a way in the integration to define them.  I also do not see the Protect NVR auto added.

**Comment by briis** [comment, marked as answer]
You cannot install the Integration from here. As said in #262 HACS is only a way to bring a non-standard integration in to HA. All this does is copy the files to a directory in Home Assistant. Below is an extract from #262 on what to do now?

So once you select Unifi Protect in HACS, and install it, the only thing that happens is that the necessary files are copied to Home Assistant. Then you need to re-start Home Assistant, for the system to pick-up the new files. At this point, Unifi protect is now part of Integrations that can be installed from the Integrations page.

But, this page often gets cached in your browser, so to ensure you can find Unifi Protect, go to Configuration and then the Integrations page, and do a hard refresh of your browser (CMD+R on a Mac).
Now press the button in the lower right corner named + ADD INTEGRATION, and search for Unifi Protect. It should now be available to configure.

**Comment by jimmarks** [comment]
AH!!!!

When you are in the HACS screen, there is an Add Integration there as well.  I understood that it added the files, but when I saw it there, I assumed it was "integrated"

The reality is that you should go to the main integrations screen and search for Unifi Protect again and add the main integration there.

Sorry for being so dense, but appreciate the extra "encouragement"

**Reply by briis** [reply]
I get the question often, so you are not alone in getting confused. Hope you will get it up and running with your VLAN. (Not a scenario I have tested)

**Reply by jimmarks** [reply]
Did it within the few seconds it took to reply to your last post.  All cameras added and viewable.  
I did add an additional Network Interface to my HA box and put it on that VLan, so that might have done it, but again, its working.  Thanks for the help

**Comment by darkrain-nl** [comment]
Don't worry that scenario will work fine it the necessary ports are open to the other VLAN ;-)
